client/landing/stepper/declarative-flow/internals/components/survery-manager/test/index.test.tsx

```diff
@@ -0,0 +1,69 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { SITE_MIGRATION_FLOW } from '@automattic/onboarding';
+import { render } from '@testing-library/react';
+import React from 'react';
+import SurveyManager from '../';
+import { type Flow } from '../../../types';
+
+jest.mock( '../../../steps-repository/components/migration-survey/async', () => {
+	const AsyncMigrationSurvey = () => 'Mock Survey';
+
+	return AsyncMigrationSurvey;
+} );
+
+jest.mock( '../../deferred-render', () => {
+	const DeferredRender = ( { children }: { children: React.ReactNode; timeMs?: number } ) =>
+		children;
+
+	return { DeferredRender };
+} );
+
+describe( 'SurveyManager', () => {
+	it( 'should render null when disabled prop is true', () => {
+		const flow: Flow = {
+			name: SITE_MIGRATION_FLOW,
+			isSignupFlow: true,
+			initialize: jest.fn(),
+			useStepNavigation: jest.fn(),
+		};
+
+		const { container } = render( <SurveyManager disabled flow={ flow } /> );
+
+		expect( container ).toBeEmptyDOMElement();
```

Comment:
Not a blocker for this PR, but let's definitely follow up on the TS errors that appear here. 

A way to ensure this works is to migrate the `setup-test-framework.js` file to TS and make sure TS sees it properly - because we already have the `import '@testing-library/jest-dom';` import there and that should be enough to resolve the issue IMO.

Reply:
> Not a blocker for this PR, but let's definitely follow up on the TS errors that appear here.

I'll plan to open a follow-up pull request. It's a little difficult to confirm which specific changes have an impact, but yes I think we'll want to include the setup file in `tsconfig.json` and/or `@testing-library/jest-dom` as `types` directly.

Reply:
As raised in Slack (p1745346000251659-slack-C02DQP0FP), the issue with the TypeScript errors in this file is partly due to the fact that we [exclude test files in type-checking](https://github.com/Automattic/wp-calypso/blob/1dd51b77a280033c1983cdf23bb2a9fdeaae1b00/client/tsconfig.json#L22), so any additional configuration to ensure the `@testing-library/jest-dom` types are loaded is ignored. Based on the Slack conversation, I think this is a longer-term effort to avoid excluding these files, which will require fixing a bunch of existing issues.